Backend/src/main/java/com/example/Backend/controller/UserController.java

```diff
@@ -0,0 +1,128 @@
+package com.example.Backend.controller;
+
+import com.example.Backend.exceptions.InvalidInputException;
+import com.example.Backend.exceptions.UserNotFoundException;
+import com.example.Backend.exceptions.UserRegistrationException;
+import com.example.Backend.model.UserLogin;
+import com.example.Backend.model.UserRegister;
+import com.example.Backend.services.LoginServiceInterface;
+import com.example.Backend.services.RegisterServiceInterface;
+import jakarta.validation.Valid;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.BindingResult;
+import org.springframework.validation.FieldError;
+import org.springframework.web.bind.annotation.*;
+
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import io.swagger.v3.oas.annotations.responses.ApiResponses;
+
+import java.util.List;
+
+@Slf4j
+@RestController
+public class UserController {
+
+    RegisterServiceInterface registerServiceInterface;
+    LoginServiceInterface loginServiceInterface;
+
+    UserController(RegisterServiceInterface registerServiceInterface, LoginServiceInterface loginServiceInterface) {
+        this.registerServiceInterface = registerServiceInterface;
+        this.loginServiceInterface = loginServiceInterface;
+    }
+
+    @Operation(summary = "Display user details", description = "Get user details by username")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "400", description = "Empty Username"),
+            @ApiResponse(responseCode = "404",description = "User Doesn't exists"),
+            @ApiResponse(responseCode = "200", description = "User Fetched Successfully")
+    })
+    @GetMapping("/user/{name}")
+    public ResponseEntity<?> displayUser(@PathVariable(value = "name") String username) {
+        UserLogin userLogin = this.loginServiceInterface.getUserDetails(username);
+        switch (StringUtils.isEmpty(username)?"Empty": userLogin.getUsername()) {
```

Comment:
No need to use a switch here. The exception will be thrown from the loginService functions. you can use the throws keyword for that function. Once springboot see the required exception it will automatically call the controller advice. Your controller class will remain short and simple :)